cppcache/src/TcrConnection.cpp

```diff
@@ -173,7 +173,22 @@ bool TcrConnection::initTcrConnection(
     ACE_OS::hostname(hostName, sizeof(hostName) - 1);
 
     ACE_INET_Addr driver(hostName);
-    uint32_t hostAddr = driver.get_ip_address();
+    uint8_t hostaddr[16];
+    int len;
+    if (driver.get_type() == AF_INET6) {
+      const struct sockaddr_in6* sa6 =
+          static_cast<const struct sockaddr_in6*>(driver.get_addr());
+      auto saddr = reinterpret_cast<const uint8_t*>(&sa6->sin6_addr);
+      len = sizeof(sa6->sin6_addr);
+      memcpy(hostaddr, saddr, len);
```

Comment:
This is where I would just put the host adder into the vector so there is no question of ownership later.